specs/fault-proof/stage-one/anchor-state-registry.md

```diff
@@ -0,0 +1,450 @@
+and incorrect game resolution.
+
+#### Dependencies
+
+- [aASR-003](#aasr-003-the-anchorstateregistry-will-be-correctly-initialized-at-deployment)
+- [aFDG-002](#afdg-002-fault-dispute-games-with-correct-claims-resolve-correctly-at-some-regular-rate)
+
+### iASR-005: The anchor game is a game whose claim is correct.
+
+When asked for the **anchor game** by fault dispute games, the contract will only serve an **anchor game** that is a
+valid game or a valid-retired game.
+
+#### Impact
+
+**Severity: High**
+
+If this invariant is broken, dispute games can be created with incorrect starting points, leading to games that can be
+used to prove false claims. This would lead to an operational failure, requiring incident response. If incident response
+doesn't occur, this could lead to loss of funds.
+
+#### Dependencies
+
+- TODO
+
+## Function-Level Invariants
+
+## Implementation Spec
+
+### `constructor`
+
+The constructor must disable the initializer on the implementation contract.
+
+### `initialize`
+
+- Initial anchor state must be an **authorized input**.
+- Dispute game factory must be an **authorized input**.
+- `dispute game finality delay` must be an **authorized input**.
+- Superchain config must be an **authorized input**.
+
+### `getRecentValidGame`
+
+Returns **anchor game**. Reverts if the **anchor game** has been **retired**.
+
+### `updateAnchorGame`
+
+- Game must be a **valid game**.
+- Game must not be a **retired game**.
+- Game's block number must be higher than current **anchor game**.
+- This function is the ONLY way to update the **anchor game** (after initialization).
+
+### `getAnchorGame`
+
+Returns the **anchor game**.
+
+- Must revert if the **anchor game** is blacklisted.
+- Must maintain the property that the timestamp of the game is not too old.
+  - TODO: How old is too old?
+
+### `registerLikelyValidGame`
+
+Stores the address of a **likely valid game** in an array as a candidate for **anchor game**.
+
+- Callable only by a **likely valid game**.
+- Calling game must only register itself (and not some other game).
+  - TODO: determine any invariants around registry ordering.
+
+### `tryUpdateAnchorGame`
+
+Try to update **anchor game** using registry of **likely valid games**.
+
+- Callable by anyone.
+- Find the latest (comparing on l2BlockNumber) valid game you can find in the register within a fixed amount of gas.
+  - Fixed gas amount ensures that this function does not get more expensive to call as time goes on.
+- Use this as input to `updateAnchorGame`.
+
+### `isGameBlacklisted`
+
+Returns whether the game is a **blacklisted game**.
+
+### `isGameLikelyValid`
+
+Returns whether the game is a **likely valid game**.
+
+### `isGameFinalized`
+
+Returns whether the game is a **finalized game**.
+
+### `isGameValid`
+
+Returns whether the game is a **valid game**.
+
+### `setRespectedGameType`
+
+- Must be **authorized** by guardian role.
+
+### `retireAllExistingGames`
+
+Retires all currently deployed games.
```

Comment:
there's an implication in the scenario here that i'd like to double click on:

- all games are retired
- `getAnchorGame` returns a game that's retired.
- all games will be using the retired anchor game for their root until another game resolves and waits the finality duration
- that means the anchor game won't update for at least `min(game resolution time) + finality duration delay`

are we ok with an anchor game being days old?

in the above scenario, if games get retired a second time before another game became valid, the anchor state could be up to `2 * (min(game resolution time) + finality duration delay)` seconds old. when does this become a problem?